File: 2023/8/8.py
import re

network = open("8.txt").read().splitlines()
direction = list(network[0])
network = {re.split(r"[\W]+", i)[0]: tuple(re.split(r"[\W]+", i)[1:]) for i in network[2:]}
start = "AAA"
finish = "ZZZ"
steps = 0
points = []


while True:
    for u in direction:
        steps += 1
        start = network[start][0] if u == "L" else network[start][1]
        current_point = points.append(network[start][0] if u == "L" else network[start][1])
        if start == finish:
            break
    else:
        continue
    break

print(steps)
print("This way took:", points)

File: 2023/8/8.scala
import scala.io.Source

object Main extends App {
  val st = System.currentTimeMillis()

  val network = Source.fromFile("8.txt").getLines().toSeq
  val direction = network.head.toCharArray
  val networkMap = network.tail.map(line => {
    val parts = line.split("[\\W]+")
    parts.head -> (parts(1), parts(2))
  }).toMap

  var start = "AAA"
  val finish = "ZZZ"
  var steps = 0
  var points = List[String]()

  while (true) {
    for (u <- direction) {
      steps += 1
      start = if (u == 'L') networkMap(start)._1 else networkMap(start)._2
      points :+= (if (u == 'L') networkMap(start)._1 else networkMap(start)._2)
      if (start == finish) {
        println(steps)
        println("This way according: " + points.mkString(", "))
        val et = System.currentTimeMillis()
        val elapsedTime = (et - st) / 1000.0
        println("Execution time: " + elapsedTime + " seconds")
        sys.exit()
      }
    }
  }
}